cmd/geth/chaincmd.go

```diff
@@ -394,6 +405,33 @@ func exportPreimages(ctx *cli.Context) error {
 	return nil
 }
 
+// exportOverlayPreimages dumps the preimage data to a flat file.
+func exportOverlayPreimages(ctx *cli.Context) error {
+	if ctx.Args().Len() < 1 {
+		utils.Fatalf("This command requires an argument.")
+	}
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	chain, _ := utils.MakeChain(ctx, stack)
+
+	var root common.Hash
+	if ctx.String(utils.TreeRootFlag.Name) != "" {
+		rootBytes := common.FromHex(ctx.String(utils.StartKeyFlag.Name))
+		if len(rootBytes) != common.HashLength {
+			return fmt.Errorf("invalid root hash length")
+		}
+		root = common.BytesToHash(rootBytes)
+	}
```

Comment:
@gballet, I've added an optional flag `--roothash` that can be provided to choose a specific hash to walk the MPT.
The flag will use the latest known state if it isn't provided.

This can be helpful in the future.